lib/resourceapply/cv.go

```diff
@@ -1,92 +1,66 @@
-	if required.Status.Extension.Raw != nil && required.Status.Extension.Object != nil {
-		return nil, false, fmt.Errorf("both extension.Raw and extension.Object should not be set")
-	}
-	existing, err := client.ClusterOperators(required.Namespace).Get(required.Name, metav1.GetOptions{})
+func ApplyClusterVersion(client cvclientv1.ClusterVersionsGetter, required *cvv1.ClusterVersion) (*cvv1.ClusterVersion, bool, error) {
+	existing, err := client.ClusterVersions().Get(required.Name, metav1.GetOptions{})
 	if errors.IsNotFound(err) {
-		actual, err := client.ClusterOperators(required.Namespace).Create(required)
+		actual, err := client.ClusterVersions().Create(required)
 		return actual, true, err
 	}
 	if err != nil {
 		return nil, false, err
 	}
 
 	modified := pointer.BoolPtr(false)
-	resourcemerge.EnsureOperatorStatus(modified, existing, *required)
+	resourcemerge.EnsureClusterVersion(modified, existing, *required)
 	if !*modified {
 		return existing, false, nil
 	}
 
-	actual, err := client.ClusterOperators(required.Namespace).Update(existing)
+	actual, err := client.ClusterVersions().Update(existing)
 	return actual, true, err
 }
 
-func ApplyOperatorStatusFromCache(lister oslistersv1.ClusterOperatorLister, client osclientv1.ClusterOperatorsGetter, required *osv1.ClusterOperator) (*osv1.ClusterOperator, bool, error) {
-	if required.Status.Extension.Raw != nil && required.Status.Extension.Object != nil {
-		return nil, false, fmt.Errorf("both extension.Raw and extension.Object should not be set")
-	}
-	existing, err := lister.ClusterOperators(required.Namespace).Get(required.Name)
+func ApplyClusterVersionFromCache(lister cvlistersv1.ClusterVersionLister, client cvclientv1.ClusterVersionsGetter, required *cvv1.ClusterVersion) (*cvv1.ClusterVersion, bool, error) {
+	obj, err := lister.Get(required.Name)
 	if errors.IsNotFound(err) {
-		actual, err := client.ClusterOperators(required.Namespace).Create(required)
+		actual, err := client.ClusterVersions().Create(required)
 		return actual, true, err
 	}
 	if err != nil {
 		return nil, false, err
 	}
 
 	// Don't want to mutate cache.
-	existing = existing.DeepCopy()
-	modified := pointer.BoolPtr(false)
-	resourcemerge.EnsureOperatorStatus(modified, existing, *required)
-	if !*modified {
-		return existing, false, nil
-	}
-
-	actual, err := client.ClusterOperators(required.Namespace).Update(existing)
-	return actual, true, err
-}
-
-func ApplyClusterVersion(client cvclientv1.ClusterVersionsGetter, required *cvv1.ClusterVersion) (*cvv1.ClusterVersion, bool, error) {
-	existing, err := client.ClusterVersions(required.Namespace).Get(required.Name, metav1.GetOptions{})
-	if errors.IsNotFound(err) {
-		actual, err := client.ClusterVersions(required.Namespace).Create(required)
-		return actual, true, err
-	}
-	if err != nil {
-		return nil, false, err
-	}
-
+	existing := new(cvv1.ClusterVersion)
```

Comment:
why not `DeepCopy()` ?

Reply:
This was moved older code, I will update